### ISSUE TITLE
Add null check in from() method for modifiable generated class if method is nullable Map

### ISSUE THREAD

**Issue by treenwang** [immutables/immutables#960]
The generated modifiable classes doesn't invoke null check before invoking putAllXXX() in `from()` method. The example is as below:
```import org.immutables.value.Value;

import javax.annotation.Nullable;
import java.util.Map;

@Value.Modifiable
@Value.Immutable
public interface Test {
    @Nullable Map<String, String> getMap();
}
```

The generated "from()" method in ModifiableTest class:
```  
  public ModifiableTest from(ModifiableTest instance) {
    Objects.requireNonNull(instance, "instance");
    putAllMap(instance.getMap());
    return this;
  }
```
It will throw NullPointerException if the instance.getMap() is null. The method `public ModifiableTest from(Test instance)` also need to be fixed. 
BTW, the immutable class is correct:
```
    public final Builder from(ModifiableTest instance) {
      Objects.requireNonNull(instance, "instance");
      @Nullable Map<String, String> mapValue = instance.getMap();
      if (mapValue != null) {
        putAllMap(mapValue);
      }
      return this;
    }
```

**Comment by elucash** [comment]
Thank you for reporting the issue. This should be fixed

**Comment by michaelwmason** [comment]
Is there any indication of when this could be worked? My team has a need for this to be implemented.

**Comment by elucash** [comment]
thank you for bumping it up, we can fix this in next bugfix release

**Comment by elucash** [comment]
released in 2.9.3
